### PR TITLE
New ECDH vectors

### PR DESCRIPTION
**Pull request by simo5**
Add new Vectors from NIST that include a DKM value to verify the Key Derivation Aftrer ECDH Key Exchange

### CODE REVIEW

**Comment by public** [comment]
I have confirmed that the sha256 digest of the files ` vectors/cryptography_vectors/asymmetric/ECDH/KASValidityTest_ECCStaticUnified_KDFConcat_NOKC_resp.fax` and ` vectors/cryptography_vectors/asymmetric/ECDH/KASValidityTest_ECCStaticUnified_KDFConcat_NOKC_init.fax` matches the files in the zip file currently at `http://csrc.nist.gov/groups/STM/cavp/documents/keymgmt/KASTestVectorsECC2014.zip` with SHA-256 `293f25702327dedec32205612ee4000ae5f404f9de645564e8d3ed9ebf64dfaf`.

The modifications to the files `vectors/cryptography_vectors/asymmetric/ECDH/KASValidityTest_ECCStaticUnified_NOKC_ZZOnly_init.fax` and ` vectors/cryptography_vectors/asymmetric/ECDH/KASValidityTest_ECCStaticUnified_NOKC_ZZOnly_resp.fax` don't match any vectors currently distributed by NIST though.

**Comment by simo5** [comment]
Should we update the other vectors with current NIST files ?
Be aware that doing that will remove test vectors for P-192 and P-224 as those curves are not recommended anymore therefore NIST dropped them from the 2014 test files.

Simo.

**Comment by public** [comment]
> Be aware that doing that will remove test vectors for P-192 and P-224 as those curves are not recommended anymore

Source? We should document this.

**Comment by public** [comment]
The tables in http://csrc.nist.gov/publications/nistpubs/800-131A/sp800-131A.pdf have anything >= P-224 as valid for key agreement AIUI.

**Comment by simo5** [comment]
On Sat, 2015-05-02 at 06:08 -0700, Alex Stapleton wrote:
> > Be aware that doing that will remove test vectors for P-192 and P-224 as those curves are not recommended anymore
> 
> Source? We should document this.

I simply downloaded the test vector files and older curves vectors are
gone. I do not have any "official" source but those and the fact Suite-B
only recommends P-256 and P-384.

Simo.

-- 
Simo Sorce * Red Hat, Inc * New York

**Comment by public** [comment]
Can you remove the vector loader from this PR and document in `docs/development/test-vectors.rst
` that we have "CAVS 17.4" vectors for ECDH too now. Then I am +1 on this.

**Comment by simo5** [comment]
On Sun, 2015-05-03 at 03:24 -0700, Alex Stapleton wrote:
> Can you remove the vector loader from this PR and document in `docs/development/test-vectors.rst
> ` that we have "CAVS 17.4" vectors for ECDH too now. Then I am +1 on this.

The vector loader commit is here because this PR depends on #1876, I do
not know how to tell github to not show that commit in here, it just
does, once #1876 will be merged in master I can rebase this PR so that
the first commit disappears.

I will add the text to docs/development/test-vectors.rst

Simo.

-- 
Simo Sorce * Red Hat, Inc * New York

**Comment by reaperhulk** [comment]
This can now be rebased so github will show only the one commit (and then we can verify/merge this)

**Comment by simo5** [comment]
Rebased

File: tests/test_utils.py [hunk 1]
```diff
@@ -3336,6 +3336,75 @@ def test_load_kasvs_ecdh_vectors():
     assert expected == load_kasvs_ecdh_vectors(vector_data)
 
 
+def test_load_kasvs_ecdh_kdf_vectors():
```
Comment (reaperhulk) [inline tests/test_utils.py]: Could you move this test + the small loader changes to a separate PR? (Sorry!) I'd like the added test vectors to be the only thing in this one so I can verify them and merge them independently of reviewing any code.

**Comment by simo5** [comment]
Moved actual tests changes to #1907 as requested

**Comment by reaperhulk** [comment]
I have confirmed these vectors match the ones in the `KASTestVectorsECC2014` zip file from NIST CAVP.

**Pull request merged**